### PR TITLE
JNI changes for range-extents in window functions.

### PR DESCRIPTION
**Pull request by mythrocks**
This commit adds back-end JNI changes to support explicit range-extents for ranged window functions. The change is analogous to the addition of `cudf::range_window_bounds::extent_type`, in `libcudf`. This change will allow for `STRING` order-by columns for range window functions. It is required because without it, it would be impossible to differentiate between `UNBOUNDED` and `CURRENT ROW` for a window function over a `STRING` order-by column.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/WindowOptions.java [hunk 1]
```diff
@@ -33,8 +50,10 @@ enum FrameType {ROWS, RANGE}
   private final int orderByColumnIndex;
   private final boolean orderByOrderAscending;
   private final FrameType frameType;
-  private final boolean isUnboundedPreceding;
-  private final boolean isUnboundedFollowing;
+  private final RangeExtentType precedingBoundsExtent;
+  private final RangeExtentType followingBoundsExtent;
+//  private RangeExtentType precedingBoundsExtent = RangeExtentType.BOUNDED;
+//  private RangeExtentType followingBoundsExtent = RangeExtentType.BOUNDED;
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/WindowOptions.java:56]: Need cleanup?

**Review by ttnghia — APPROVED** [review]

File: java/src/main/java/ai/rapids/cudf/WindowOptions.java (hunk 1 above)
Comment (mythrocks) [reply java/src/main/java/ai/rapids/cudf/WindowOptions.java:56]: Thank you. Yes, I should've removed that.

Comment (mythrocks) [reply java/src/main/java/ai/rapids/cudf/WindowOptions.java]: Removed now.

**Review by revans2 — APPROVED** [review]

**Pull request merged**